VideoLocker/src/main/java/org/edx/mobile/view/DownloadListActivity.java

```diff
@@ -152,26 +152,32 @@ public DownloadItem(@NonNull DownloadEntry model, @NonNull NativeDownloadModel n
         }
 
         @Override
-        public long getSize() {
-            if (model.size == 0) {
+        @Nullable
+        public Long getSizeInBytes() {
+            if (model.size <= 0) {
+                if (nativeModel.size <= 0) {
+                    return null;
```

Comment:
Hmm, instead of using a nullable auto-unboxed `Long` type as the return value, we could simply return the original value of `nativeModel.size` (as done in the previous implementation), and handle `<= 0` from the calling code. That would provide more information as well (pending vs failed), making this method technically more useful I guess.